### DISCUSSION TITLE
Removing AssemblyName from osu.Desktop.csproj enables exclusive fullscreen

### DISCUSSION THREAD

**Discussion by goodtrailer** [ppy/osu#15198] [Q&A] [answered]
Addresses #6075. TL;DR Removing the `AssemblyName` element from `osu.Desktop.csproj` fixes the issue of not entering exclusive fullscreen (on my laptop at least).

### Explanation
This isn't a PR because I feel like I'm doing something wrong or am missing something. This "fix" is just too bizarre, but it's also really consistent on my laptop. Removing the following line makes osu! enter exclusive fullscreen: https://github.com/ppy/osu/blob/6c1d22416466689cd91836743922649d998197d6/osu.Desktop/osu.Desktop.csproj#L7 Otherwise, it doesn't enter exclusive fullscreen. This happens every single time, and I'm not really sure why. I first noticed something was weird when I was unit testing and realized that `VisualTestRunner` was running in exclusive fullscreen, but `osu.Desktop` never did. I created some new projects from scratch which just ran a `new OsuGame()`, and those projects ran in exclusive fullscreen too. Took me a while to find the problem because I didn't think to look in the csproj. And yes, this works on `Release` builds too.

### Video
https://www.youtube.com/watch?v=4hTV2VziIC0

With the `AssemblyName` attribute, osu! is captured by OBS display capture, and the cursor latency is very noticeable (this happens off-recording too). With the `AssemblyName` element removed and the project rebuilt, osu! is unfortunately no longer captured by OBS because it's in exclusive fullscreen, but you can still hear the music and menu sounds.

The reason you can see the Windows cursor is because I made a new `OsuGameDesktopWithCursor` class that just unsets the `CursorState.Hidden` flag, you can see it here in my fork: https://github.com/goodtrailer/osu/blob/f0ca74ec6aef4c667d8891da3c180de6515874a5/osu.Desktop/OsuGameDesktopWithCursor.cs#L8-L22

### Laptop specs
Model: Dell Inspiron 7577
CPU: Intel Core i7-7700HQ
iGPU: Intel HD Graphics 630
dGPU: NVIDIA GeForce GTX 1060 Max-Q
OS: Windows 10 20H2 (19042.1288)
Display: 4K 60Hz, with osu! running at 1080p

### osu! version
https://github.com/goodtrailer/osu/tree/exclusive-fullscreen
Fork was created after 6c1d22416466689cd91836743922649d998197d6 (after 2021.1016.0)

**Comment by peppy** [comment, marked as answer]
Have you checked your driver settings? You may have a custom profile based on the window title.

**Reply by goodtrailer** [reply]
I did have settings set, but resetting everything to default for both titles (osu! and osu.Desktop) changed nothing, it still wouldn't open in exclusive fullscreen.

**Reply by goodtrailer** [reply]
These were the settings after clicking restore, if they're useful:
https://user-images.githubusercontent.com/46171121/138012654-ab816842-213a-460a-ab3c-3e672f323252.png
https://user-images.githubusercontent.com/46171121/138012669-ea64beea-8792-4a82-a54e-4ca3cc0d1c29.png
https://user-images.githubusercontent.com/46171121/138012681-2b7952e2-0185-41e9-acd8-fe29e2067847.png
https://user-images.githubusercontent.com/46171121/138012695-dc559e0b-f42a-49b4-a9aa-8717931c10f4.png

**Reply by goodtrailer** [reply]
Nevermind, I've figured it out, I *was* missing something obvious (?). Windows graphics settings are not window title based, and NVIDIA's are. Yeah, I'm dumb, sorry.

**Reply by peppy** [reply]
Great to hear. Which setting was it that was causing an issue? Might be handy if this comes up again for other users.

**Reply by goodtrailer** [reply]
No it wasn't a specific setting, I just had osu! and osu.Desktop both set to integrated through Windows graphics settings without realizing that because the osu! was the actual one (the one in appdata and not bin), the setting wasn't affecting builds, which is why it wasn't launching with the integrated gpu. So even though I thought they both had the same gpu running, they weren't, so that's why one was working fine with exclusive fullscreen and the other wasn't.
